cvat/apps/lambda_manager/views.py

```diff
@@ -0,0 +1,307 @@
+            url = NUCLIO_GATEWAY
+
+        reply = getattr(requests, method)(url, headers=NUCLIO_HEADERS,
+            timeout=NUCLIO_TIMEOUT, json=data)
+        reply.raise_for_status()
+        response = reply.json()
+
+        return response
+
+    def list(self):
+        data = self._http(url=self.NUCLIO_ROOT_URL)
+        response = [LambdaFunction(self, item) for item in data.values()]
+        return response
+
+    def get(self, id):
+        data = self._http(url=self.NUCLIO_ROOT_URL + '/' + id)
+        response = LambdaFunction(self, data)
+        return response
+
+    def invoke(self, func, payload):
+        return self._http(method="post", port=func.port, data=payload)
+
+class LambdaFunction:
+    def __init__(self, gateway, data):
+        # ID of the function (e.g. omz.public.yolo-v3)
+        self.id = data['metadata']['name']
+        # type of the function (e.g. detector, interactor)
+        self.kind = data['metadata']['annotations'].get('type')
+        # dictionary of labels for the function (e.g. car, person)
+        spec = json.loads(data['metadata']['annotations'].get('spec') or '[]')
+        labels = [item['name'] for item in spec]
+        if len(labels) != len(set(labels)):
+            raise ValidationError(
+                "`{}` lambda function has non-unique labels".format(self.id),
+                code=status.HTTP_404_NOT_FOUND)
+        self.labels = labels
+        # state of the function
+        self.state = data['status']['state']
+        # description of the function
+        self.description = data['spec']['description']
+        # http port to access the serverless function
+        self.port = data["status"]["httpPort"]
+        # framework which is used for the function (e.g. tensorflow, openvino)
+        self.framework = data['metadata']['annotations'].get('framework')
+        # display name for the function
+        self.name = data['metadata']['annotations'].get('name')
+        self.gateway = gateway
+
+    def to_dict(self):
+        response = {
+            'id': self.id,
+            'kind': self.kind,
+            'labels': self.labels,
+            'state': self.state,
+            'description': self.description,
+            'framework': self.framework,
+            'name': self.name,
+        }
+        return response
+
+    def invoke(self, db_task, frame, points=None):
+        payload = {
+            'image': self._get_image(db_task, frame),
+            'points': points
+        }
+
+        return self.gateway.invoke(self, payload)
+
+    def _get_image(self, db_task, frame):
+        frame_provider = FrameProvider(db_task.data)
+        # FIXME: now we cannot use the original quality because nuclio has body
+        # limit size by default 4Mb (from FastHTTP).
+        image = frame_provider.get_frame(frame, quality=FrameProvider.Quality.COMPRESSED)
+        return base64.b64encode(image[0].getvalue()).decode('utf-8')
+
+
+class LambdaQueue:
+    def _get_queue(self):
```

Comment:
Codacy found an issue: [Method could be a function](https://app.codacy.com/manual/cvat/cvat/pullRequest?prid=5729179)